### PR TITLE
Drop support for Python 3.6, 3.7

### PR DESCRIPTION
**Pull request by Jongy**
After https://github.com/Granulate/granulate-utils/pull/107 is merged.

Since https://github.com/Granulate/gprofiler/pull/597 we no longer need any 3.6, 3.7 support. Time to move forward

### CODE REVIEW

**Comment by Jongy** [comment]
```
Version 3.10.0 was not found in the local cache
Error: Version 3.10.0 with arch x[6](https://github.com/Granulate/gprofiler/actions/runs/3649238071/jobs/6163646479#step:2:7)4 not found
The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

Pfft

**Comment by Jongy** [comment]
This seems to explain - https://github.com/actions/setup-python/issues/555 they are upgrading `ubuntu-latest` to 22.04 so now we start getting breakages.

**Comment by Jongy** [comment]
Trying 3.10.x which allegedly (per #255 at leat) PyPerf doesn't support. Let's see..

File: .github/workflows/build-container-and-test.yml [hunk 1]
```diff
@@ -57,12 +60,15 @@ jobs:
...
+    runs-on: ubuntu-20.04  # for 3.10.0 python tag
 
     strategy:
       fail-fast: false  # helps detecting flakiness / errors specific to one Python version
       matrix:
-        python-version: [3.6, 3.7, 3.8.8, 3.9.2, 3.10.0]
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10.0"
```
Comment (Jongy) [inline .github/workflows/build-container-and-test.yml:71]: Comment with reference to pyperf bug ticket

File: gprofiler/profilers/python_ebpf.py [hunk 2]
```diff
@@ -216,8 +216,7 @@ def _dump(self) -> Path:
             # using read1() which performs just a single read() call and doesn't read until EOF
             # (unlike Popen.communicate())
             assert self.process is not None
-            # Python 3.6 doesn't have read1() without size argument :/
-            logger.debug(f"PyPerf output: {self.process.stderr.read1(4096)}")  # type: ignore
+            logger.debug(f"PyPerf output: {self.process.stderr.read1()}")  # type: ignore
```
Comment (Jongy) [inline gprofiler/profilers/python_ebpf.py:219]: Fix other `read1` sites

File: gprofiler/profilers/registry.py [hunk 3]
```diff
@@ -2,8 +2,7 @@
 
 
 class ProfilerArgument:
-    # It would have been better to replace with a dataclass, but since we want to support Python 3.6 this is the best
-    # alternative as we still need the get_dict method (which is not convenient to do with a namedtuple)
+    # TODO convert to a datalcass
```
Comment (d3dave) [inline gprofiler/profilers/registry.py:5]:
```suggestion
    # TODO convert to a dataclass
```

**Review by d3dave — DISMISSED** [review]

**Comment by Jongy** [comment]
I still don't understand the latest CI failure, debugging

**Comment by Jongy** [comment]
The last change seems to have "solved" it.

**Review by d3dave — APPROVED** [review]

**Pull request merged**